packages/xo-server-test/src/backupNg/backupNg.spec.js

```diff
@@ -389,4 +444,114 @@ describe('backupNg', () => {
     })
     expect(vmTask.data.id).toBe(vmId)
   })
+
+  test('execute three times a delta backup with 2 remotes, 2 as retention and 2 as fullInterval', async () => {
+    jest.setTimeout(6e4)
+    await xo.createTempServer(config.servers.default)
+    const { nfs, smb } = config.remotes
+    const { id: nfsId } = await xo.createTempRemote(nfs)
+    const remotes = [nfsId]
+    if (smb !== undefined) {
+      const { id: smbId } = await xo.createTempRemote(smb)
+      remotes.push(smbId)
+    }
+
+    const scheduleTempId = randomId()
+    const vmId = config.vms.vmToBackup
+    const fullInterval = 2
+    const { id: jobId } = await xo.createTempBackupNgJob({
+      mode: 'delta',
+      remotes: {
+        id: {
+          __or: remotes,
+        },
+      },
+      schedules: {
+        [scheduleTempId]: DEFAULT_SCHEDULE,
+      },
+      settings: {
+        '': {
+          reportWhen: 'never',
+          fullInterval,
+        },
+        [nfs.id]: { deleteFirst: true },
+        [scheduleTempId]: { exportRetention: 2 },
+      },
+      vms: {
+        id: vmId,
+      },
+    })
+
+    const schedule = await xo.getSchedule({ jobId })
+    expect(typeof schedule).toBe('object')
+
+    const nExecutions = 3
+    const backupsByRemote = await xo.runBackupJob(jobId, schedule.id, {
+      remotes,
+      nExecutions,
+    })
+    forOwn(backupsByRemote, backups => expect(backups.length).toBe(2))
+
+    const backupLogs = await xo.call('backupNg.getLogs', {
+      jobId,
+      scheduleId: schedule.id,
+    })
+    expect(backupLogs.length).toBe(nExecutions)
+
+    backupLogs.forEach(({ tasks, ...log }, key) => {
+      validateRootTask(log, {
+        data: {
+          mode: 'delta',
+          reportWhen: 'never',
+        },
+        message: 'backup',
+        status: 'success',
+      })
+      let vmTaskValidated = false
```

Comment:
You can store the number of tasks of type `VM, snapshot, transfer ...` and validate it at the end.